BrainPortal/app/models/userfile.rb

```diff
@@ -727,15 +728,30 @@ def initialize_from_hash(atts = {}) #:nodoc:
         cb_error "Invalid :if condition '#{method}' in model." unless method.respond_to?(:to_proc)
         @conditions << method.to_proc
       end
+      @errors = []
     end
 
     def valid_for?(userfile) #:nodoc:
       return true if @conditions.empty?
-      @conditions.all? { |condition| condition.call(userfile) }
+
+      @conditions.all? do |condition|
+        [condition.call(userfile)].flatten.all? { |error| error == true}
+      end
     rescue
       false
     end
 
+    def apply_conditions(userfile) #:nodoc:
+      errors = @conditions.map do |condition|
+        condition_call = condition.call(userfile)
+        condition_call = (condition_call.is_a?(TrueClass) ? [] :
+                ( condition_call.is_a?(FalseClass) ? [false] :
+                  condition_call
+                )).flatten
+      end.flatten
+      errors.each {|e| self.errors << e if self.errors.exclude?(e)}
```

Comment:
Weird... ? Does it mean `self.errors = (self.errors + errors).uniq`  ?

Reply:
Also, given that errors is a attr_accessor, the internal variable shoudl be named something else, because the version first `errors=` at the first line could be interpreted as a method call